lib/Raven/Serializer.php

```diff
@@ -67,17 +67,44 @@ public function __construct($mb_detect_order = null)
      * sanitization and encoding.
      *
      * @param mixed $value
+     * @param array $context
+     * @return string|bool|double|int|null|object|array
+     */
+    public function serialize($value, array $context = [])
+    {
+        $full_context = $this->get_full_context($context);
+        return $this->serialize_inner($value, $full_context['max_depth'], 0);
+    }
+
+    /**
+     * @param array $context
+     * @return array
+     */
+    protected function get_full_context(array $context)
+    {
+        if (!array_key_exists('max_depth', $context)) {
```

Comment:
Okay. 

1. You do `composer require sentry/sentry`
2. You see what serializers are not working how you want them to work
3. You write your own Serializer
4. By any reasons you want to change serializers on the fly. I.e. you write your own warning capturing in your own CMS. You want to change standard serializer for global Raven client and you do not want to extend \\Raven\\Client. What do you do without setters and reflections? Only extend \\Raven\\Client and override `capture` where you change currents serializer and call `parent::capture`

I want get users their own choice to change behaviour of Raven.